src/hmr.fs

```diff
@@ -3,10 +3,102 @@ namespace Elmish.HMR
 open Elmish
 open Elmish.Browser
 open Fable.Import
+open Fable.Core.JsInterop
 
 [<RequireQualifiedAccess>]
 module Program =
 
+    [<RequireQualifiedAccess>]
+    module internal Cmd =
+        /// Execute the commands using the supplied dispatcher
+        let inline exec (dispatch:Dispatch<'msg>) (cmd:Cmd<'msg>)=
+            cmd |> List.iter (fun sub -> sub dispatch)
+
+    type RunMsg<'msg> =
+        | UserMsg of 'msg
+        | Stop
+
+    type RunModel<'model> =
+        | Inactive
+        | Active of 'model
+
+    #if DEBUG
+    /// Start the program loop.
+    /// arg: argument to pass to the init() function.
+    /// program: program created with 'mkSimple' or 'mkProgram'.
+    let inline runWith (arg: 'arg) (program: Program<'arg, 'model, 'msg, 'view>) =
+
+        let map (model, cmd) =
+            model, cmd |> Cmd.map UserMsg
+
+        let update (msg : RunMsg<'msg>) (model : RunModel<'model>) =
+            match msg with
+            | UserMsg msg ->
+                match model with
+                | Inactive -> model, Cmd.none
+                | Active userModel ->
+                    let newModel, cmd = program.update msg userModel
+                    Active newModel, cmd
+            | Stop ->
+                Inactive, Cmd.none
+            |> map
+
+        let setState (model : RunModel<'model>) dispatch =
+            match model with
+            | Inactive -> ()
+            | Active userModel ->
+                program.setState userModel (UserMsg >> dispatch)
+
+        let (model,cmd) =
+            let userModel, cmd = program.init arg
+            Active userModel, Cmd.map UserMsg cmd
+
+        let inbox = MailboxProcessor.Start(fun (mb:MailboxProcessor<RunMsg<'msg>>) ->
```

Comment:
I didn't expect to see this reimplemented here. This could become a problem, so I think we need to find a way not to reimplement the loop.